### PR TITLE
Auto filter tests

### PR DESCRIPTION
**Pull request by drewharvey**
Removed all listeners inside auto filter component and instead used observer to watch for value changes.  Adding some basic tests for the auto filter component.

### CODE REVIEW

File: px-auto-filter-field.html [hunk 1]
```diff
@@ -85,45 +85,27 @@
...
-              type: String
+              type: String,
+              observer: '_onChange'
             }
           };
         }
 
         ready() {
           super.ready();
-          const input = this.shadowRoot.querySelector('input');
```
Comment (limonte) [inline px-auto-filter-field.html]: I'm confused, as PR title states you're adding tests, not changing `px-auto-filter-field` component. Are changes in `px-auto-filter-field.html` intentional? If so, you need to move them to separated PR or at least to separated commit(s).

File: test/px-data-grid-test.js [hunk 2]
```diff
@@ -189,5 +208,60 @@ function runTests() {
...
+    describe('auto filter field tests', () => {
+
+      beforeEach(() => {
+        grid = fixture('grid-with-auto-filter');
+        grid.tableData = data;
+        grid.autoFilter = true;
+      });
+
+      it('should show only 1 row after filtering', (done) => {
+        Polymer.RenderStatus.afterNextRender(grid, () => {
```
Comment (limonte) [inline test/px-data-grid-test.js]: DRY: move this line and line before to `beforeEach`, remove copy-pasted stuff from all test cases inside this `describe` section.

**Comment by limonte** [comment]
Sign CLA.

File: px-auto-filter-field.html (hunk 1 above)
Comment (alump) [reply px-auto-filter-field.html]: Ok, this is fine, titles to be improvised in the future

**Pull request merged**